satellite/gracefulexit/endpoint.go

```diff
@@ -204,6 +204,14 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 		endpoint.connections.delete(nodeID)
 	}()
 
+	isDisqualified, err := endpoint.handleDisqualifiedNode(ctx, nodeID)
+	if err != nil {
+		return rpcstatus.Error(rpcstatus.Internal, err.Error())
+	}
+	if isDisqualified {
```

Comment:
I think maybe this check and return could just be done inside of handleDisqualified, so it only ends up returning an error.

Reply:
Yea, I was debating between returning an `ErrDisqualifiedNode` vs returning a boolean for the check. Do you think an specific error would be better?